custom/icds/management/commands/populate_mother_name.py

```diff
@@ -0,0 +1,214 @@
+import csv
+import six
+import sys
+import time
+
+from datetime import (
+    datetime,
+    date,
+    timedelta,
+)
+from xml.etree import cElementTree as ElementTree
+
+from django.core.management.base import BaseCommand
+
+from corehq.apps.users.util import SYSTEM_USER_ID
+from corehq.form_processor.backends.sql.dbaccessors import CaseAccessorSQL, CaseReindexAccessor
+from corehq.form_processor.exceptions import CaseNotFound
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.apps.locations.models import SQLLocation
+from corehq.apps.hqcase.utils import submit_case_blocks
+from corehq.form_processor.backends.sql.dbaccessors import iter_all_rows
+
+from casexml.apps.case.mock import CaseBlock
+
+DOMAIN = "icds-cas"
+CASE_TYPE = "person"
+CUT_OFF_AGE_IN_YEARS = 6
+date_today = date.today()
+CUT_OFF_DOB = str(date_today.replace(year=date_today.year - CUT_OFF_AGE_IN_YEARS))
+
+DOB_PROPERTY = "dob"
+MOTHER_NAME_PROPERTY = "mother_name"
+MOTHER_INDEX_IDENTIFIER = "mother"
+
+CASE_ITERATION_COUNT = 10000
+MAX_RESCUE_EXCEPTIONS_ON_UPDATE = 5
+
+CSV_HEADERS = ['Case ID', 'Mother Case ID', 'Mother Name']
+
+TEST_STATES = []
+
+for loc in SQLLocation.active_objects.filter(location_type__code='state', domain=DOMAIN):
+    if loc.metadata.get('is_test_location') == 'test':
+        TEST_STATES.append(loc.name)
+
+
+class Command(BaseCommand):
+    help = """
+    Iterate person cases updated in last 100 days (3 months with buffer) in a single partition,
+    Find the ones which are
+        - not deleted
+        - not belonging to test locations,
+        - with age less than 6 years using dob case property,
+            - if there is related mother case, populate mother_name case property with it's name
+    Returns two lists of case ids, the ones updated and the ones that could not be updated
```

Comment:
Firstly, Two files generated
1. list of cases ids that for which we could find the mother name
2. list of case ids where we could not find the mother name

the last file is to keep track of the updates as they are happening necessary in case the updates fails mid way and needs to be restarted.

Reply:
I see.